packages/core-components/src/components/OAuthRequestDialog/OAuthRequestDialog.tsx

```diff
@@ -58,6 +62,12 @@ export function OAuthRequestDialog(_props: {}) {
   const classes = useStyles();
   const [busy, setBusy] = useState(false);
   const oauthRequestApi = useApi(oauthRequestApiRef);
+  const configApi = useApi(configApiRef);
+  const usePopup = configApi.getOptionalBoolean('auth.usePopup') ?? true;
+  const redirectMessage = usePopup
+    ? ''
+    : 'This will trigger a http redirect to OAuth Login.';
```

Comment:
There is an assumption here that the auth APIs are instantiated in a way that reads the the config value, which might not always be the case. I could see that someone for example might want to use redirect flow on the sign-in page for a single provider, but popup for the rest. You can do that by overriding the default implementation for the auth API used for sign-in.

Do you think it would be a slim change to communicate this through the OAuthRequest API instead? Perhaps we can include the auth mode in the request for example?